frontend/src/metabase/query_builder/components/NativeQueryEditor.jsx

```diff
@@ -526,6 +546,13 @@ class NativeQueryEditor extends Component {
 
 export default _.compose(
   ExplicitSize(),
+  Questions.loadList({
```

Comment:
Locally I'm seeing 404s to `/api/card//referenced-cards?endpoint=referenced-cards` where the `:questionId` param is missing (and the `"referenced-cards"` string is used as a query, but that isn't the source of the error), I think because this renders before we've fetched the underlying card object associated with the id in the url, so `props.question` isn't ready to be used as intended.

`Questions.loadList` is coupled to the `fetchList` action added to entities, and `fetchList` generally (or maybe in every other scenario) does not need args to work, so unlike `Question.load({ entityId: ... })`, it won't wait for the presence of values. Not a bad idea to reuse `loadList` for this in my opinion, but we'll need to change some things in the `EntityListLoader` component for it to work. One possibility off the top of my head (surely there are other ideas) -- add an additional property to `loadList` that accepts `props` and returns a predicate to determine whether or not we are ready to trigger the action and send the query.

(Also, assuming you get `loadList` to work, add `loadingAndErrorWrapper: false` to this config object to prevent this component from messing with UI stuff on this page)

Reply:
The reason the app seems to work still is that there's a UI component on the page that is wrapped in `Question.load(...)` -- the sidebar that pops out, so the card is still fetched regardless of the above request failing.

Reply:
Now that I am thinking about it more, we won't always have a `question.id()` available, so it likely makes more sense to collect a list card ids using a method like the one above (`referencedQuestionIds`) and use it to trigger `Questions.actions.fetch({ entityId: cardId })` calls. Going via the `fetch` action ensures the FE caches the card so that the GET doesn't get triggered over and over.

In order to use `Questions.actions.fetch` in a component you will need to dispatch it via redux. See an old example here: https://github.com/metabase/metabase/blob/8b16f94ee80fc354663b56beac00182f0132ee36/frontend/src/metabase/containers/SavedQuestionLoader.jsx#L135

Reply:
> Locally I'm seeing 404s to /api/card//referenced-cards?endpoint=referenced-cards where the :questionId param is missing

Huh. That's not what was happening for me, so there must be a race condition. Anyway, it proves the point.

> we won't always have a question.id() available,

For my understanding, under what condition will we not have a question.id() available?

> use it to trigger Questions.actions.fetch({ entityId: cardId }) calls

Do we have any examples of collecting results from a bunch of calls like this, that I could take a look at? 

An alternative approach would be to create a new endpoint that returns a list of questions at once, e.g. `Questions.actions.fetchList({ entityIds: commaSeparatedQuestionIds })`. This is what Ngoc suggested above.